maestro-driver-ios/maestro-driver-iosUITests/Routes/Handlers/SubTreeRouteHandler.swift

```diff
@@ -14,15 +14,25 @@ class SubTreeRouteHandler: RouteHandler {
         
         let viewHierarchyHttpResponse = await MainActor.run { () -> HTTPResponse in
             do {
+                let springboardBundleId = "com.apple.springboard"
                 logger.info("Trying to capture hierarchy snapshot for \(appId)")
                 let start = NSDate().timeIntervalSince1970 * 1000
                 let xcuiApplication = XCUIApplication(bundleIdentifier: appId)
-                let springboardApplication = XCUIApplication(bundleIdentifier: "com.apple.springboard")
-                if appId != "dev.mobile.maestro-driver-iosUITests.xctrunner" && springboardApplication.buttons["Allow"].exists {
-                    xcuiApplication.tap()
-                }
-                logger.info("Now trying hierarchy")
-                let viewHierarchyDictionary = try xcuiApplication.snapshot().dictionaryRepresentation
+                let springboardApplication = XCUIApplication(bundleIdentifier: springboardBundleId)
+                
+                logger.info("[Start] Now trying hierarchy for: \(appId)")
+                var viewHierarchyDictionary = try xcuiApplication.snapshot().dictionaryRepresentation
+                logger.info("[Done] Now trying hierarchy for: \(appId)")
+                logger.info("[Start] Now trying hierarchy for: \(springboardBundleId)")
+                let springboardHierarchyDictionary = try springboardApplication.snapshot().dictionaryRepresentation
+                logger.info("[Done] Now trying hierarchy for: \(springboardBundleId)")
+                
+                let children = viewHierarchyDictionary[XCUIElement.AttributeName(rawValue: "children")] as! Array<[XCUIElement.AttributeName: Any]>
```

Comment:
is force type cast fine here, what if one of them returns an empty response of smth ?

Reply:
Can you suggest how can we handle that here?

Reply:
make both optional, soft-fail if one of them is missing